proto/backend.proto

```diff
@@ -255,3 +255,33 @@ message StreamPacket {
   // JSON-encoded data to publish into a channel.
   bytes data = 1;
 }
+
+message MigrationsRequest {
+  enum SchemaType {
+    DATASOURCE_QUERY = 0;
+  }
+
+  // The type of data we are migrating
+  SchemaType schemaType = 1;
+
+  // SemVer compatible version string
+  string schemaVersion = 2;
+  
+  // JSON-encoded data to migrate
+  bytes data = 3;
+}
+
+message MigrationsResponse {
+  enum Status {
+    OK = 0;
+    ERROR = 1;
+  }
```

Comment:
Protobufs really don't need to define error types, as the generated servers stubs are expected to `return nil, err` or throw an error in an error case

Reply:
For example, for a `RequestDenied` error you would define an error type in Go, `var ErrorRequestDenied = errors.New("request denied")`, and return that in your server implementation.

Reply:
More examples of gRPC error handling here: https://github.com/avinassh/grpc-errors/tree/master/go

:)

Reply:
👍 thanks. i copied the status pattern from the other responses, and didn't consider that